indexd/index/drivers/alchemy.py

```diff
@@ -1096,6 +1097,28 @@ def get(self, did, expand=True):
 
             return record.to_document_dict()
 
+    def get_with_nonstrict_prefix(self, did, expand=True):
+        """
+        Attempt to retrieve a record both with and without a prefix.
+        Proxies 'get' with provided id.
+        If not found but prefix matches default, attempt with prefix stripped.
+        If not found and id has no prefix, attempt with default prefix prepended.
+        """
+        try:
+            # Try to resolve id as provided
+            record = self.get(did, expand=expand)
+        except NoRecordFound as e:
+            DEFAULT_PREFIX = self.config.get("DEFAULT_PREFIX")
+            match = re.match(r"(dg\.[0-9a-f]{4}\/)(.+)", did)
```

Comment:
@paulineribeyre do we only want to attempt resolution if the prefix matches this specific format? or should we just see if the provided id starts with whatever `DEFAULT_PREFIX` is set to, even if it's not spec compliant?

Reply:
i would limit it to the configured `DEFAULT_PREFIX` (we call it default, but i'm pretty sure it's actually the only allowed prefix)